mitodl/ocw-studio
# Cc/publish alert

#### Pre-Flight checklist


#### What are the relevant tickets?
Finishes #991 

#### What's this PR do?
A few things:
- adds a custom alert prompt instead of window.confirm, see screenshot below 
- control opening the publishing drawer via a search param `?publish=`. Its value is irrelevant. Its key present means "drawer is open".
- Use the `?publish=` search param along with react-router's prompt component to prompt user for confirmation when publishing with a dirty form (*this was the ticket goal ✅*)
- Adds a new frontend integration test helper to live alongside the old. See code comment.

#### How should this be manually tested?
1. Restart docker to pick up new dependencies. *Possibly only need to restart watch: `docker-compose restart watch`*.
2. Try publishing with unsaved changes on the metadata page (or menu page; any page with editable content where publish button is still visible). You should be prompted for confirmation if and only if you have unsaved changes.
3. You should still be prompted when navigating to other routes with a dirty for
4. You should not be prompted in other scenarios.

#### Screenshots (if appropriate)

<img width="546" alt="Screen Shot 2022-04-25 at 10 23 37 AM" src="https://user-images.githubusercontent.com/9010790/165109087-bc42af61-c2c3-460e-abd0-da5d02076cf6.png">
